rfcs/0112-demote-darwin-support-tier.md

```diff
@@ -0,0 +1,91 @@
+---
+feature: demote_darwin_support_tier
+start-date: 2021-10-27
+author: piegames
+co-authors: many
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Move x86_64-darwin from a Tier 2 platform to Tier 3, as described in [RFC 0046](https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md).
+
+# Motivation
+[motivation]: #motivation
+
+Darwin is constantly adding additional maintainer burden. Especially, it does not live up to the requirements:
+
+- "A lot of packages built by Hydra, full ofBorg support."
+    - There is only one ofBorg builder. It can take several hours or days for an ofBorg build to finish for Darwin, if it finishes at all.
+    - Hydra builders a greatly understaffed.
+    - If Apple follows through on [their announcement to discontinue](https://www.businessinsider.com/apple-macbook-pro-discontinued) x86-based devices, the situation won't improve due to a physical lack of hardware.
```

Comment:
It's possible to get `x86_64-darwin` with https://github.com/lhotari/action-upterm and other github actions for a period of 6h at once.

Reply:
“It is possible” is anything but “it is easy and well supported”. Do you have detailed instructions on how to do that with little overhead? e.g. how to integrate it into nix as a remote builder?

Reply:
If the solution is (ab)using GitHub actions in a way that is clearly not intended and may well be prohibited in the future, then it may rather prove the point above?

Reply:
I think in the future `x86_64-darwin` will definitely become hard to support, but that will happen as `aarch64-darwin` will become more common and easier to support. We shouldn't justify demoting the highest tier darwin platform with architecture specific reasoning alone. Demoting `x86_64-darwin` to tier 3 while `aarch64-darwin` is still tier 3 is completely different from doing that when `aarch64-darwin` is tier 2.

Reply:
I have a quite straightforward solution to get an x86_64 macos machine spun up for ~$0.12/h if anyone wants to ping me about it.

Reply:
@risicle ping